### PR TITLE
Add RubyOnRails example

### PR DESCRIPTION
**Pull request by domenkozar**
cc @bobvanderlinden

### CODE REVIEW

File: examples/rubyonrails/.test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+set -ex
+rails new blog -d=postgresql
+devenv up&
+timeout 20 bash -c 'until echo > /dev/tcp/localhost/5100; do sleep 0.5; done'
+pushd blog
+    rails db:create
+popd
```
Comment (bobvanderlinden) [inline examples/rubyonrails/.test.sh:8]: CI is failing for `pushd`. I guess `sh` doesn't support it.

```suggestion
(cd blog && rails db:create)
```

**Comment by domenkozar** [comment]
macOS:

> /Users/runner/work/devenv/devenv/examples/rubyonrails/.test.sh: line 5: timeout: command not found

**Comment by bobvanderlinden** [comment]
Ugh, maybe the tests should get a separate nix file that is used on top of the devenv.nix in the example. Include all of the environment that is needed for the test in there (bash, curl, etc)?

**Comment by domenkozar** [comment]
I've added a section that injects the packages.

Maybe we could also add helper functions for testing, but that's for later.

**Pull request merged**